src/main/java/com/metsci/laproc/application/Application.java

```diff
@@ -19,15 +21,18 @@
     /** The Toolbox that provides all features in this application */
     private ToolBox globalToolBox;
     /** A global reference to the relevant data that is used by Tools */
-    private DataReference globalDataReference;
+    private InputDataReference globalInputDataReference;
+    /** A global reference to the relevant data that is used by Tools */
```

Comment:
23 and 25 should be different. Maybe relevant input data and relevant output data